### ISSUE TITLE
[Bug] System.NullReferenceException at Xamarin.Forms.Platform.Android.ShellFlyoutRecyclerAdapter.GenerateItemList()

### ISSUE THREAD

**Issue by Basti82b** [xamarin/Xamarin.Forms#6846]
### Description

After the reset of an apps Shell it's not possible to add new items.

### Steps to Reproduce

1. Reset an apps Shell: Application.MainPage = new AppShell();
2. Try to add items to the new instance.

### Actual Behavior

![image](https://user-images.githubusercontent.com/49757194/60974270-1521c780-a32a-11e9-9ff6-8f12a7ba53e7.png)

See: https://github.com/xamarin/Xamarin.Forms/blob/4.1.0/Xamarin.Forms.Platform.Android/Renderers/ShellFlyoutRecyclerAdapter.cs#L169

### Basic Information

- Version with issue: Xamarin.Forms 4.1.0.555618
- Platform Target Frameworks: 
  - Android: 9.0

**Comment by samhouts** [comment]
@Basti82b  Can you please attach a small project that demonstrates this issue? Thanks!

**Comment by Basti82b** [comment]
@samhouts 
While creating a sample project, I found out that the problem is caused by the GC not (completely) destroying the Shell object because it subscribes to events.
The error message appears because the old shell object is trying to add entries to itself in the event handler.
So I implemented the IDisposable interface.
Maybe I should use the MessagingCenter instead of events.
I apologize for my ignorance.
